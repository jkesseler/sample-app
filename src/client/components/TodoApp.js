import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import CssBaseline from '@material-ui/core/CssBaseline';
import { loadTodos } from '../../state/todos/actions';
import Layout from './Layout';
import AddTodo from './AddTodo';
import TodoList from './TodoList';


class TodoApp extends Component {
  componentDidMount() {
    const { dispatch } = this.props;
    dispatch(loadTodos());
  }

  render() {
    return (
      <CssBaseline>
        <Layout>
          <AddTodo />
          <TodoList />
        </Layout>
      </CssBaseline>
    );
  }
}


TodoApp.propTypes = {
  dispatch: PropTypes.func.isRequired,
};


export default connect(state => ({ todos: state.todos }))(TodoApp);
