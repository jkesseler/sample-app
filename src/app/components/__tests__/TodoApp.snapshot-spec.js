import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import TodoApp from '../TodoApp';

describe('TodoApp', () => {
  it('renders without crashing', () => {
    const component = shallow(<TodoApp />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
