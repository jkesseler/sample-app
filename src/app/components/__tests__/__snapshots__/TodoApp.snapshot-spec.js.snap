// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoApp renders without crashing 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
